arkouda/timeclass.py

```diff
@@ -144,6 +143,158 @@ def __init__(self, array, unit: str = _BASE_UNIT):  # type: ignore
+            days = cast((seconds // 86400), int64)
+            self._year = cast((days // 365.24 + 1970), int64)
+            self._month = cast(((seconds % 31556926) / 2629743), int64) + 1
+            self._day = cast((((seconds % 31556926) % 2629743) / 86400), int64) + 2
+
+            self._hour = cast((self.values / _get_factor("h")), int64)
+            self._hour %= 24
+            self._minute = cast((self.values / _get_factor("m")), int64)
+            self._minute %= 60
+            self._second = cast((self.values / _get_factor("s")), int64)
+            self._second %= 60
+
+            # Month, day
+            self._day = self.values // _get_factor("d")
+            self._day += 1401  # move starting point to 1996 - 03 - 01
+            self._day %= 1461  # max 3 years 365 day (not 4 years!) remaining
+            d = self._day[:]
+            self._day[d >= 731] -= 731
+
+            d = self._day[:]
+            self._day[d >= 365] -= 365
+
+            d = self._day[:]
+            self._month = d[:] - d[:] + 3
+            self._day[d >= 306] -= 306
+            self._month[d >= 306] = 1
+
+            d = self._day[:]
+            self._day[d >= 153] -= 153
+            self._month[d >= 153] = 8
+
+            d = self._day[:]
+            self._day[d >= 122] -= 122
+            self._month[d >= 122] += 4
+            d = self._day[:]
+            self._day[d >= 61] -= 61
+            self._month[d >= 61] += 2
+
+            d = self._day[:] + 2
+            self._month[d >= 32] += 1
+            months = self._month[:]
+            months[months >= 13] -= 1
+            self._month[self._month >= 13] = 1
+            self._day += 2
+            self._day[d >= 32] -= 31
+
+            # Day of year
+            month_days = [
+                [0, 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334],
+                [0, 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335],
+            ]
+
+            lyear1 = self._year % 4 == 0
+            lyear2 = self._year % 100 != 0
+            lyear3 = self._year % 400 == 0
+            lyear = lyear1 & lyear2
+            lyear = cast((lyear | lyear3), int64)
+
+            days = self._day[:]
+            self._date = []
+            for i in range(days.size):
+                if self._month[i] == 2 and self._day[i] == 29 and lyear[i] == 0:
+                    self._month[i] += 1
+                    self._day[i] = 1
+                    days[i] += month_days[lyear[i]][self._month[i] - 1]
+                else:
+                    days[i] += month_days[lyear[i]][self._month[i]]
+                self._date.append(f"{self._year[i]}-{self._month[i]:02d}-{self._day[i]:02d}")
+            self._dayofyear = ak.array(days)
+
+            # Day of week
+            self._dayofweek = cast(((self.values / _get_factor("d")) + 3), int64)
+            self._dayofweek %= 7
+
+            # Week of year
+            week = (self._dayofyear[:] - 1) - self._dayofweek + 3
```

Comment:
`self._dayofyear[:]` is making an unnecessary copy of `self._dayofyear`. On large arrays, this will be an issue. Update this to `week = (self._dayofyear - 1) - self._dayofweek + 3`.

Reply:
This unnecessary copy is still present